elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -133,21 +179,57 @@ public static void main(String[] args) {
      */
     public void readAndValidateConfigs() throws IOException {
         this.loadConfigMap();
-        this.setModelVariables(readVariableConfig(Config.MODELVARIABLE));
-        this.setElideSecurityConfig(readSecurityConfig());
-        validateRoleInSecurityConfig(this.elideSecurityConfig);
-        this.setDbVariables(readVariableConfig(Config.DBVARIABLE));
+        this.modelVariables = readVariableConfig(Config.MODELVARIABLE);
+        this.elideSecurityConfig = readSecurityConfig();
+        validateSecurityConfig();
+        this.dbVariables = readVariableConfig(Config.DBVARIABLE);
         this.elideSQLDBConfig.setDbconfigs(readDbConfig());
         this.elideTableConfig.setTables(readTableConfig());
         validateRequiredConfigsProvided();
         validateNameUniqueness(this.elideSQLDBConfig.getDbconfigs());
         validateNameUniqueness(this.elideTableConfig.getTables());
         validateInheritance(this.elideTableConfig);
         populateInheritance(this.elideTableConfig);
-        validateTableConfig(this.elideTableConfig);
+        validateTableConfig();
         validateJoinedTablesDBConnectionName(this.elideTableConfig);
     }
 
+    public void hydrateAndCompileModelConfigs() throws Exception {
+        hydrateAndCompileModelConfigs(ElideDynamicInMemoryCompiler.newInstance().ignoreWarnings());
+    }
+
+    public void hydrateAndCompileModelConfigs(ElideDynamicInMemoryCompiler compiler) throws Exception {
+        HandlebarsHydrator hydrator = new HandlebarsHydrator(staticModelDetails);
+        Map<String, String> tableClasses = hydrator.hydrateTableTemplate(this.elideTableConfig);
+        Map<String, String> securityClasses = hydrator.hydrateSecurityTemplate(this.elideSecurityConfig);
+
+        compiler.useParentClassLoader(getClass().getClassLoader());
```

Comment:
Seems like this belongs elsewhere (like maybe in the compiler constructor).

Reply:
moved.